renderer/rst.go

```diff
@@ -0,0 +1,149 @@
+package renderer
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/elastic/crd-ref-docs/config"
+	"github.com/elastic/crd-ref-docs/types"
+)
+
+const (
+	rstAnchorPrefix = ""
+)
```

Comment:
No it's probably not :) - As you can probably tell - I copy pasted it :)
I am using Sphinx to build it, perhaps it would be better named SphinxRST?
I did not yet sign the Elastic CLA, I need approval from my company first - I was unaware it was a requirement when I submitted the PR

Reply:
I pointed the tool at the [ECK project](https://github.com/elastic/cloud-on-k8s) (this tool was built to support it) and ran the output through Sphinx. I have never used Sphinx before, so I please tell me if I am using it incorrectly. Some of the errors I saw are:

```
sphinx-build -b singlehtml -C . ~/tmp

Running Sphinx v3.5.0
...
out.rst:67: WARNING: Error with CSV data in "csv-table" directive:
',' expected after '"'

.. csv-table::
   :header: "Field", "Description"
   :widths: 10, 40

   "``version`` (string)", "Version of the Agent."
   "``elasticsearchRefs`` (:ref:`Output` array)", "ElasticsearchRefs is a reference to a list of Elasticsearch clusters running in the same Kubernetes cluster. Due to existing limitations, only a single ES cluster is currently supported."
   "``image`` (string)", "Image is the Agent Docker image to deploy. Version has to match the Agent in the image."
   "``config`` (:ref:`Config`)", "Config holds the Agent configuration. At most one of [`Config`, `ConfigRef`] can be specified."
   "``configRef`` (:ref:`ConfigSource`)", "ConfigRef contains a reference to an existing Kubernetes Secret holding the Agent configuration. Agent settings must be specified as yaml, under a single "agent.yml" entry. At most one of [`Config`, `ConfigRef`] can be specified."
   "``secureSettings`` (:ref:`SecretSource`)", "SecureSettings is a list of references to Kubernetes Secrets containing sensitive configuration options for the Agent. Secrets data can be then referenced in the Agent config using the Secret's keys or as specified in `Entries` field of each SecureSetting."
   "``serviceAccountName`` (string)", "ServiceAccountName is used to check access from the current resource to a Elasticsearch resource in a different namespace. Can only be used if ECK is enforcing RBAC on references."
   "``daemonSet`` (:ref:`DaemonSetSpec`)", "DaemonSet specifies the Agent should be deployed as a DaemonSet, and allows providing its spec. Cannot be used along with `deployment`."
   "``deployment`` (:ref:`DeploymentSpec`)", "Deployment specifies the Agent should be deployed as a Deployment, and allows providing its spec. Cannot be used along with `daemonSet`."
...
out.rst:222: WARNING: Duplicate explicit target name: "apmserver".
...
out.rst:243: WARNING: Duplicate explicit target name: "apmserverspec".
...
```

The CLA is a recent organizational change. I'll update the README to indicate its presence.